src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -845,47 +855,119 @@ QgsExpressionContext QgsLayoutItemLegend::createExpressionContext() const
 
   context.appendScope( scope );
 
-  return context;
+  if ( replace )
+  {
+    mExpContext.~QgsExpressionContext();
```

Comment:
What does this do exactly?

Reply:
This aims to reset the current expression context using the proper method instead of creating a new one and replacing the member.

Reply:
Moved. to new class